model/map/src/main/java/org/keycloak/models/map/user/MapUserEntity.java

```diff
@@ -44,6 +44,8 @@ abstract class AbstractUserEntity extends Impl implements MapUserEntity {
 
         private static final Logger LOG = Logger.getLogger(MapUserProvider.class);
         private String id;
+        private String username;
+        private String usernameLowerCase;
```

Comment:
Each implementation should be able to decide which way it would implement lettercase (in)sensitive search. For that reason, there should be no lowercase username variant prescribed in the generic map entity (nor in the AbstractUserEntity).

If that is needed say in JPA, then JPA entity metadata object should extend the `AbstractUserEntity` and add lowercase variant there. Yet - can we do without it?

Reply:
> Yet - can we do without it?

It's not possible to create generated column from already generated column, therefore I've added the the field into `JpaUserEntity`. Or did you have something else in mind?